transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java

```diff
@@ -182,7 +188,13 @@ private boolean loginMissingToken(NextFilter nextFilter,
             }
 
             loginContext.login();
+        } catch (AccessControlLoginException ace) {
+            if (loggerEnabled()) {
+                log("Login failed: ", ace);
+            }
 
+            writeResponse(HttpStatus.CLIENT_FORBIDDEN, nextFilter, session, httpRequest);
+            return false;
         } catch (LoginException le) {
             // Depending on the login modules configured, this could be
             // a very normal condition.
```

Comment:
Currently, Gateway sends 403 in the `catch (LoginException le)` clause only if the `loginContext` is null. However, in the case of InetAddressFilterLoginModule, the `loginContext` will not be null and Gateway still has to send a 403 when the remote IP address does not match the ones that are specified in the config. That's why I created `AccessControlLoginException` that will allow the Gateway to send a 403 regardless of `loginContext`'s value.